### DISCUSSION TITLE
Invoking zoom programmatically under UWP

### DISCUSSION THREAD

**Discussion by nirdil** [MicrosoftEdge/WebView2Feedback#3964] [Q&A]
I'm looking for a way to invoke zoom on a webview2 programmatically.

Under UWP webview2 does not expose the `CoreWebView2Controller.ZoomFactor` function.

I tried to use the DevTools Protocol to invoke the zoom using the `setPageScaleFactor` function, which works well in terms of zooming but I can't get the scrollbars to appear on zooming in. Any ideas?

**Comment by champnic** [comment]
@krschau Any ideas on this one? Is there a way to do this in the WinUI controls?

@nirbil This functionality would have to be exposed by the WinUI controls themselves, as the CoreWebView2Controller is not exposed directly in the WinUI controls (nor the .NET controls).
